### ISSUE TITLE
Feature: Support SQlite and use it as default database

### ISSUE THREAD

**Issue by sazarkin** [daffyyyy/CS2-SimpleAdmin#91]
It will be nice to get sqlite integrated database out of the box without extra MySQL instance required.

Database can be an interface and have two implementation: sqlite and mysql.

I play maybe once a month on my server and do not want to keep mysql instance only for ban/mute logic to work.

**Comment by Dliix66** [comment]
The main target with this plugin is to be able to share the DB accross server and/or access its data through a web plugin or discord bot.
This would not be possible with SQLite.
We *could* make an SQLite option, but this should never be the default.

**Comment by sazarkin** [comment]
> but this should never be the default.

I mean if you have not put mysql settings in config, it should use sqlite.

Right now if you haven't configured mysql half of features do not work.

**Comment by jstnmthw** [comment]
> This would not be possible with SQLite.

Of course it would be possible with SQLite. Why wouldn't it?

Running a SQLite docker image with a persistent volume comes to mind, for example.

**Comment by daffyyyy** [comment]
> > This would not be possible with SQLite.
> 
> Of course it would be possible with SQLite. Why wouldn't it?
> 
> Running a SQLite docker image with a persistent volume comes to mind, for example.

Nope, sqlite allow only 1 connection to write at once, u can read but write = bottleneck

**Comment by daffyyyy** [comment]
> > Nope, sqlite allow only 1 connection to write at once, u can read but write = bottleneck
> 
> Bottleneck != not possible.
>  
> Besides in real world scenario you might achieve somewhere between write 20k writes per second. How often are you writing to your database? And even if you [were](https://www.sqlite.org/wal.html).
> 

Penalties with timemode set to 0 update database every minute for each player. 
When player connect with new nickname but banned, then plugin update nickname/ip in database.
When player connect then plugin save ip to database (if new IP) even if not banned.
Connect = map change too

**Comment by kus** [comment]
+1 to SQLite support

I just want to use a nice admin plugin, with a menu, to manage my 1 server that my mates play on, and make it easy to kick/ban people that join and are toxic.
